### ISSUE TITLE
Tests running(?) with errors @ RC6

### ISSUE THREAD

**Issue by Cayan** [angular/quickstart#208]
When I run the tests I get the following output:

https://gist.github.com/Cayan/3e684dbe40b14cb3acdce00122bcf043

**Comment by psnider** [comment]
same here for commit-id 73f4a820f271e9463103dfb52f92989abfa9c284

on Mac OSX 10.11.6
[output of npm list](https://gist.github.com/psnider/119cee8139a2c4e568b064494272b973)

**Comment by Foxandxss** [comment]
Yes, we are syncing the testing chapter with the quickstart so this will be fixed for next week.

**Comment by filipesilva** [comment]
Tests seem to be running correctly with the latest quickstart, are you using the new SystemJS and karma config?

**Comment by Cayan** [comment]
Yes, this issue was created with SystemJS and Karma.

I figured it out, it's actually is a silly bug. The error reported is a socket error, Karma tries to run on the defined port and on failure we have something like:

`Chrome 52.0.2743 (Linux 0.0.0): Executed 0 of 3 SUCCESS (0 secs / 0 secs)`
`Missing error handler on socket.`

This can be easily reproduced if you start `npm run test`, stop/pause the process and attempts to run it again.

I saw somewhere else a Karma config that auto increments the port until it picks a free port.

**Comment by filipesilva** [comment]
I guess we could auto discover the port, but if the problem is that the port isn't available... then the solution is to make that port available. Adding port auto-discovery is a non-essential optimization imho, and this is a quickstart.

Unless there's a strong argument otherwise, I think this issue is closed.

**Comment by timofeysie** [comment]
I am having a similar error with @Cayan, my gist is here:<script src="https://gist.github.com/timofeysie/3cf46e2fffea24e6887ade811f14ad16.js"></script>
In the karma-test-shim.js file, it seems like the dist directory is wrong:
```
var builtPath = '/base/app/';
...
System.config({
  baseURL: '/base',
```
Get rid of the base part, since all our compiled files are next to their ts files in the app directory.
However, the karma.conf.js file also had the js path set wrong, it appears:
```
  var testBase    = 'testing/';       // transpiled test JS and map files
  var testSrcBase = 'testing/';       // test source TS files
```
After looking at the commit for #212, I removed all the testBase/testSrcBase references, but I'm still getting the error shown in the gist.  I used the QuickStart starting on Fri Sep  9 for RC6.
The problem only arose when I tried to use templateUrl and compile the templates for tests which were previously passing.
What is the solution to this issue?
